### ISSUE TITLE
Sum of stability scores as a measure of cluster quality?

### ISSUE THREAD

**Issue by chaturv3di** [scikit-learn-contrib/hdbscan#196]
I am working with dataset from a domain of which I'm not an expert. This means that I have neither labels nor ground truth clustering available to me. I am trying to figure out a good way in which I could assess the quality of clusters which I obtain upon varying the minimum number of samples and minimum cluster size.

Clustering metrics which I am aware of either assume that points in clusters are normally distributed, e.g. Silhouette Score and F-Score, or rely on a reference/ground truth labeling, e.g. adjusted mutual information and completeness scores.

I wonder if one could define a quality metric using stability scores of the flattened clusters (which I believe are available from `cluster_persistence_` field). If my understanding is correct, these scores are already _normalised_ for the sizes of different clusters since they are computed over all points assigned to these clusters. Therefore, a straightforward sum of all stability scores seems like a reasonable definition of such a metric. The higher the sum, the better the clustering; and one can search in the hyper-parameter space to find a maximal value.

It would be great to hear your thoughts on this.

**Comment by lmcinnes** [comment]
It is a measure of how stable the given clusters are, but that isn't necessarily the same thing as clustering quality -- you could, in principle, have a very stable mis-clustering. All that said, I do believe that this is a not unreasonable metric all things considered. I don't think it carries the same theoretical weight as the other options, but if you just want something to work in practice it may be "good enough".
